src/mem/cache/tags/partitioning_policies/max_capacity_pp.cc

```diff
@@ -62,32 +63,35 @@ MaxCapacityPartitioningPolicy::MaxCapacityPartitioningPolicy
             "capacities arrays are not equal lengths");
     }
 
-    // calculate total cache block count to use when creating allocation maps
-    const uint64_t total_block_cnt = this->cacheSize / this->blkSize;
-
     // check allocations and create map
     for (auto i = 0; i < this->partitionIDs.size(); i++) {
         const uint64_t partition_id = this->partitionIDs[i];
         const double cap_frac = capacities[i];
 
-        // check Capacity Fraction (cap_frac) is actually a fraction in [0,1]
-        if (!(cap_frac >= 0 && cap_frac <= 1)) {
-            fatal("MaxCapacity Partitioning Policy for PartitionID %d has "
-                "Capacity Fraction %f outside of [0,1] range", partition_id,
-                cap_frac);
-        }
-
-        const uint64_t allocated_block_cnt = cap_frac * total_block_cnt;
-        partitionIdMaxCapacity.emplace(partition_id, allocated_block_cnt);
-
-        DPRINTF(PartitionPolicy, "Configured MaxCapacity Partitioning Policy "
-            "for PartitionID: %d to use portion of size %f (%d cache blocks "
-            "of %d total)\n", partition_id, cap_frac, allocated_block_cnt,
-            total_block_cnt);
-
+        // Configure partition
+        configurePartition(partition_id, cap_frac);
     }
 }
 
+void
+MaxCapacityPartitioningPolicy::configurePartition(uint64_t partition_id,
+                                                  double cap_frac)
+{
+    // check Capacity Fraction (cap_frac) is actually a fraction in [0,1]
+    panic_if(!(cap_frac >= 0 && cap_frac <= 1),
```

Comment:
I think this should be `fatal_if` since the user provides the `cap_frac`

Reply:
So far you are right, but the purpose behind this PR is actually to allow guest SW to change the cap_frac at runtime (by programming a partitioning controller)